### PR TITLE
Use Dependency Injection

### PR DESCRIPTION
**Pull request by robert-mihaiAdam**

### CODE REVIEW

File: Domain/Dto/Bets.cs [hunk 1]
```diff
@@ -10,12 +10,17 @@ public class Bets
...
         [Required]
         public Guid BetableEntityA { get; set; }
 
         [Required]
         public Guid BetableEntityB { get; set; }
+
+        public void SetTime(TimeProvider timeProvider)
+        {
+            Date = timeProvider.GetUtcNow().DateTime;
+        }
```
Comment (victorgugonatu) [inline Domain/Dto/Bets.cs]: let's remove this - we can set the property from the service

File: Domain/Dto/PlacedBets.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ﻿using System.ComponentModel.DataAnnotations;
+using static System.Runtime.InteropServices.JavaScript.JSType;
```
Comment (victorgugonatu) [inline Domain/Dto/PlacedBets.cs]: ??? what is this ?

File: Services/BetQuoteServices.cs [hunk 3]
```diff
@@ -7,13 +7,15 @@
 
 namespace Services
 {
-    public class BetQuoteServices : IBetableService<BetQuotes, UpdateBetQuotes>
+    public sealed class BetQuoteServices : IBetQuoteService
```
Comment (victorgugonatu) [inline Services/BetQuoteServices.cs]:
```suggestion
    public sealed class BetQuoteService : IBetQuoteService
```
let's keep it in singular

File: Services/BetsServices.cs [hunk 4]
```diff
@@ -7,18 +7,22 @@
...
-        public BetsServices(DBContext dbContext)
+        public BetsServices(DBContext dBContext, TimeProvider timeProvider)
         {
-            _dbContext = dbContext;
+            Console.WriteLine("Test vtm din Bets Service cu time provider");
+            _dbContext = dBContext;
+            _timeProvider = timeProvider;
         }
 
         private async Task<bool> validateBetBody(Bets entity)
```
Comment (victorgugonatu) [inline Services/BetsServices.cs]:
```suggestion
        private async Task<bool> ValidateBetBody(Bets entity)
```

also let's append Async to all async methods

File: Services/BetsServices.cs [hunk 5]
```diff
@@ -7,18 +7,22 @@
...
         {
-            BetableEntityServices service = new(_dbContext);
+            entity.SetTime(_timeProvider);
```
Comment (victorgugonatu) [inline Services/BetsServices.cs]:
```suggestion
            entity.Date = _timeProvider.GetUtcNow().DateTime;
```

File: Domain/Dto/Bets.cs (hunk 1 above)
Comment (robert-mihaiAdam) [reply Domain/Dto/Bets.cs]: It's private the constructor

File: BetsApi/Controllers/BetsController.cs [hunk 6]
```diff
@@ -1,35 +1,43 @@
...
     public class BetsController : Controller
     {
-        private readonly DBContext _context;
+        private readonly IBetsService betService;
+        private readonly IBetQuoteService betQuoteService;
+        private readonly IPlacedBetsService placedBetsService;
+        private readonly IBetableEntityService betableEntityService;
 
-        public BetsController(DBContext context)
+        public BetsController(DBContext context,
```
Comment (victorgugonatu) [inline BetsApi/Controllers/BetsController.cs]: DBContext is not needed

File: Services/BetQuoteService.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+using DataAccess;
+using Microsoft.EntityFrameworkCore;
+using Domain.Dto;
+
+namespace Services
+{
+    public sealed class BetQuoteService : IBetQuoteService
+    {
+        private readonly DBContext _dbContext;
+        private readonly TimeProvider _timeProvider;
```
Comment (victorgugonatu) [inline Services/BetQuoteService.cs]: is this used?

File: Services/BetableEntityService.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+using DataAccess;
+using Microsoft.EntityFrameworkCore;
+using Domain.Dto;
+
+namespace Services
+{
+    public sealed class BetableEntityService : IBetableEntityService
+    {
+        private readonly DBContext _dbContext;
+        private readonly TimeProvider _timeProvider;
```
Comment (victorgugonatu) [inline Services/BetableEntityService.cs]: is this used?

File: Services/BetsService.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        public BetsService(DBContext dBContext, TimeProvider timeProvider)
+        {
+            _dbContext = dBContext;
+            _timeProvider = timeProvider;
+        }
+
+    private async Task<bool> ValidateBetBodyAsync(Bets entity)
+        {
+            entity.Date = _timeProvider.GetUtcNow().DateTime;
+            BetableEntityService service = new(_dbContext, _timeProvider);
```
Comment (victorgugonatu) [inline Services/BetsService.cs]: please inject the service instead of creating the instance manually

File: Services/PlacedBetsService.cs [hunk 10]
```diff
@@ -6,51 +6,47 @@
...
         }
 
-        public async Task<PlacedBets?> Create(PlacedBets entity)
+        public async Task<PlacedBets> CreateAsync(PlacedBets entity)
         {
+            entity.PlacedDate = _timeProvider.GetUtcNow().DateTime;
             Guid quoteId = entity.QuoteId;
-            BetQuoteServices betQuoteService = new BetQuoteServices(_dbContext);
-            BetQuotes currentQuote = await betQuoteService.GetById(quoteId);
+            BetQuoteService betQuoteService = new BetQuoteService(_dbContext, _timeProvider);
```
Comment (victorgugonatu) [inline Services/PlacedBetsService.cs]: inject the service via ctor

File: Seeder/Program.cs [hunk 11]
```diff
@@ -15,55 +15,55 @@ public static void Main(string[] args)
...
-                context.BetableEntity.Add(entity);
-            }
-            context.SaveChanges();
+            //for (long i = 1; i <= noEntities; i++)
+            //{
+            //    BetableEntityService service = new(context, TimeProvider.System);
+            //    BetableEntity entity = new BetableEntity { Name = $"Echipa {i * 12}" };
+            //    context.BetableEntity.Add(entity);
+            //}
+            //context.SaveChanges();
```
Comment (victorgugonatu) [inline Seeder/Program.cs]: is this code needed or not?
if not, delete it

**Review by victorgugonatu — APPROVED** [review]

**Pull request merged**